main.rb

```diff
@@ -0,0 +1,149 @@
+require './student'
+require './teacher'
+require './book'
+require './rental'
+
+def create_student(age, name)
+  print('Has Parent Permission [y/n]:')
+  permission = gets.chomp
+  if permission == 'y'
+    Student.new(age, name)
+  else
+    Student.new(age, name, parent_permission: false)
+  end
+end
+
+def create_teacher(age, name)
+  print('Specialization:')
+  specialization = gets.chomp
+  Teacher.new(age, specialization, name)
+end
+
+def create_person
+  print('Do you want to create a student(1) or a Teacher (2) [Input the number]:')
+  input = gets.chomp
+  print('age:')
+  age = gets.chomp
+  print('Name:')
+  name = gets.chomp
+  case input
+  when '1'
+    @persons << create_student(age, name)
+  when '2'
+    @persons << create_teacher(age, name)
+  else
+    puts('Wrong Input')
+    create_person
+  end
+  puts('Person Created succefully')
+end
+
+def create_book
+  print('title:')
+  title = gets.chomp
+  print('Author:')
+  author = gets.chomp
+  @books << Book.new(title, author)
+  puts('Book Created succefully')
+end
+
+def list_books
+  i = 0
+  @books.each do |book|
+    puts("#{i}) title: #{book.title}, author: #{book.author}")
+    i += 1
+  end
+end
+
+def list_persons
+  i = 0
+  @persons.each do |person|
+    if person.is_a?(Teacher)
+      print '[Teacher] '
+    else
+      print '[Student] '
+    end
+    puts("#{i}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}")
+    i += 1
+  end
+end
+
+def person_object(id)
+  @persons.each do |person|
+    return person if person.id == id
+  end
+  nil
+end
+
+def print_rentals
+  print('Person ID: ')
+  id = gets.chomp.to_i
+  person = person_object(id)
+  if person.nil?
+    puts('Wrong Id')
+  elsif person.rentals == []
+    puts('No rentals for this person')
+  else
+    person.rentals.each do |rent|
+      puts("Date: #{rent.date}, Book: #{rent.book.title} by #{rent.book.title}")
```

Comment:
- [ ] There's an error when displaying a rental of a person.  Kindly change the second `book.title` to something like:
`rent.author.name` so  that it will read `book title, by the author_name`

**Error:**
![image](https://user-images.githubusercontent.com/12745474/147189250-547c2826-72db-42d5-9914-cff9926dcbff.png)